tools/ctl/ctl.go

```diff
@@ -216,6 +217,47 @@ var getResults = &cobra.Command{
 	},
 }
 
+type PackageWorker interface {
+	Run(ctx context.Context, bar *pb.ProgressBar, out chan schema.Verdict, in chan benchmark.Package)
+}
+
+type WorkerPool struct {
+	Size   int
+	Worker PackageWorker
+}
+
+func (pool *WorkerPool) Process(ctx context.Context, packages []benchmark.Package) []schema.Verdict {
+	jobs := make(chan benchmark.Package, *maxConcurrency)
+	results := make(chan schema.Verdict)
+	bar := pb.StartNew(len(packages))
+	bar.ShowTimeLeft = true
+	go func() {
+		for _, p := range packages {
+			jobs <- p
+		}
+		close(jobs)
+	}()
+	var wg sync.WaitGroup
+	for i := 0; i < *maxConcurrency; i++ {
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			pool.Worker.Run(ctx, bar, results, jobs)
+		}()
+	}
+	var verdicts []schema.Verdict
+	for v := range results {
+		verdicts = append(verdicts, v)
+	}
+	wg.Wait()
+	bar.Finish()
+	log.Printf("Completed rebuilds for %d artifacts...\n", len(packages))
+	sort.Slice(verdicts, func(i, j int) bool {
```

Comment:
Not sure this belongs here. it'd simplify things to return a channel instead

Reply:
Done